aicsimageio/tests/readers/test_tiff_reader.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import pytest
+
+from aicsimageio import exceptions
+from aicsimageio.readers import TiffReader
+
+from ..conftest import LOCAL, REMOTE, get_resource_full_path
+from .reader_test_utils import run_image_read_checks, run_multi_scene_image_read_checks
+
+
+@pytest.mark.parametrize("host", [LOCAL, REMOTE])
+@pytest.mark.parametrize(
+    "filename, "
+    "set_scene, "
+    "expected_scenes, "
+    "expected_shape, "
+    "expected_dtype, "
+    "expected_dims_order, "
+    "expected_channel_names",
+    [
+        ("s_1_t_1_c_1_z_1.ome.tiff", 0, (0,), (325, 475), np.uint16, "YX", None),
+        ("s_1_t_1_c_1_z_1.tiff", 0, (0,), (325, 475), np.uint16, "YX", None),
+        (
+            "s_1_t_1_c_10_z_1.ome.tiff",
+            0,
+            (0,),
+            (10, 1736, 1776),
+            np.uint16,
+            "CYX",
+            ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"],
+        ),
+        (
+            "s_1_t_10_c_3_z_1.tiff",
+            0,
+            (0,),
+            (10, 3, 325, 475),
+            np.uint16,
+            "TCYX",
+            ["0", "1", "2"],
+        ),
+        (
+            "s_3_t_1_c_3_z_5.ome.tiff",
+            0,
+            (0, 1, 2),
+            (5, 3, 325, 475),
+            np.uint16,
+            "ZCYX",
+            ["0", "1", "2"],
+        ),
+        (
+            "s_3_t_1_c_3_z_5.ome.tiff",
+            1,
+            (0, 1, 2),
+            (5, 3, 325, 475),
+            np.uint16,
+            "ZCYX",
+            ["0", "1", "2"],
+        ),
+        (
+            "s_3_t_1_c_3_z_5.ome.tiff",
+            2,
+            (0, 1, 2),
+            (5, 3, 325, 475),
+            np.uint16,
+            "ZCYX",
+            ["0", "1", "2"],
+        ),
+        (
+            "s_1_t_1_c_1_z_1_RGB.tiff",
+            0,
+            (0,),
+            (7548, 7548, 3),
+            np.uint16,
+            "YXS",  # S stands for samples dimension
+            None,
+        ),
+        (
+            "s_1_t_1_c_2_z_1_RGB.tiff",
+            0,
+            (0,),
+            (2, 32, 32, 3),
+            np.uint8,
+            "CYXS",  # S stands for samples dimension
```

Comment:
I'm a little confused by this. above it looked like the "Samples per pixel" were being incorporated into the channel but this confuses me.

Reply:
Where above does it look like "Samples per pixel" were being incorporated into the channel?

The "Samples" dimension in `tifffile` spec is the RGB dimension. This leaves the "Channels" dimension to act "normally".